packages/ui/src/components/Notice/Notice.tsx

```diff
@@ -0,0 +1,359 @@
+import {
+  ComponentPropsWithoutRef,
+  createContext,
+  forwardRef,
+  ReactNode,
+  use,
+  useCallback,
+} from "react";
+import { tv, VariantProps } from "tailwind-variants";
+import { twMerge } from "tailwind-merge";
+import XMarkIcon from "@heroicons/react/20/solid/XMarkIcon";
+import { useIntl } from "react-intl";
+import BellAlertIcon from "@heroicons/react/24/outline/BellAlertIcon";
+import CheckCircleIcon from "@heroicons/react/24/outline/CheckCircleIcon";
+import ExclamationCircleIcon from "@heroicons/react/24/outline/ExclamationCircleIcon";
+import ExclamationTriangleIcon from "@heroicons/react/24/outline/ExclamationTriangleIcon";
+
+import { uiMessages } from "@gc-digital-talent/i18n";
+
+import useControllableState from "../../hooks/useControllableState";
+import { HeadingRank, IconType } from "../../types";
+import Separator from "../Separator";
+import IconButton, { IconButtonProps } from "../Button/IconButton";
+
+type DivProps = ComponentPropsWithoutRef<"div">;
+
+const root = tv({
+  base: "group relative grid-cols-2 gap-x-3 rounded-lg p-6 has-[>svg]:grid",
+  variants: {
+    mode: {
+      inline: "border",
+      card: "bg-white shadow-xl dark:bg-gray-700",
+    },
+    color: {
+      gray: "",
+      primary: "",
+      secondary: "",
+      success: "",
+      warning: "",
+      error: "",
+    },
+    small: {
+      true: "p-3 text-sm has-[>svg]:grid-cols-[calc(var(--spacing)*6)_1fr]",
+      false: "p-6 has-[>svg]:grid-cols-[calc(var(--spacing)*7)_1fr]",
+    },
+  },
+  compoundVariants: [
+    {
+      mode: "inline",
+      color: "gray",
+      class:
+        "bg-gray-100/20 text-gray-600 dark:bg-gray-700/20 dark:text-gray-200",
+    },
+    {
+      mode: "inline",
+      color: "primary",
+      class:
+        "bg-primary-100/20 text-primary-600 dark:bg-primary-700/20 dark:text-primary-200",
+    },
+    {
+      mode: "inline",
+      color: "secondary",
+      class:
+        "bg-secondary-100/20 text-secondary-600 dark:bg-secondary-700/20 dark:text-secondary-200",
+    },
+    {
+      mode: "inline",
+      color: "success",
+      class:
+        "bg-success-100/20 text-success-600 dark:bg-success-700/20 dark:text-success-200",
+    },
+    {
+      mode: "inline",
+      color: "warning",
+      class:
+        "bg-warning-100/20 text-warning-600 dark:bg-warning-700/20 dark:text-warning-200",
+    },
+    {
+      mode: "inline",
+      color: "error",
+      class:
+        "bg-error-100/20 text-error-600 dark:bg-error-700/20 dark:text-error-200",
+    },
+  ],
+});
+
+type RootVariants = VariantProps<typeof root>;
+
+interface NoticeContextValue extends RootVariants {
+  onDismiss?: () => void;
+}
+
+const NoticeContext = createContext<NoticeContextValue>({
+  mode: "inline",
+  color: "gray",
+  small: false,
+});
+
+export interface NoticeProps extends RootVariants, Omit<DivProps, "color"> {
+  defaultOpen?: boolean;
```

Comment:
I'm not sure I love managing the state and visibility in the component itself.  Hoisting state and all that.  It would be simpler to just emit the event to the parent and let the parent decide what to do with it and handle conditional rendering, I think.  Maybe the parent will save that in a cookie or localStorage, or just useState.  Maybe the parent will coordinate that dismiss with Notice instances on several pages.  I don't think individual components know enough to do much useful work here.